libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -283,7 +283,10 @@ func (k *kubernetesAnnotator) Run(event *beat.Event) (*beat.Event, error) {
 	kubeMeta.Delete("kubernetes.container.id")
 	kubeMeta.Delete("kubernetes.container.runtime")
 	kubeMeta.Delete("kubernetes.container.image")
+
+	orchestrator := genOrchestratorFields(kubeMeta)
 	event.Fields.DeepUpdate(kubeMeta)
+	event.Fields.DeepUpdate(orchestrator)
```

Comment:
When beats are run in GKE, `add_cloud_metadata` adds the `orchestrator.cluster.name/url` fields, but I think that only if the `orchestrator` field doesn't exist (unless fields are flattened when they reach [here](https://github.com/elastic/beats/blob/84b4cea96755e64a024f998fa836f18574f30e7a/libbeat/processors/add_cloud_metadata/add_cloud_metadata.go#L133)). We will have to check how well both things play together.

Same problem may exist with the autodiscover provider.

For context, the name/url fields were added in https://github.com/elastic/beats/pull/26056 and https://github.com/elastic/beats/pull/26438.

Reply:
@jsoriano  Thank you for reviewing.  And you've verified my suspicion that I should pull the `orchestrator` parts out and do that is a separate PR.